Feat/card-navigation-history.v1

suchachevskaya opened the pull request:
Константы навигационных элементов и маршрутов вынесены в отдельный модуль constants.ts, расположенный рядом с компонентом Header. 
Создана логика перехода внутри карточки так, чтобы при клике осуществляется навигация на страницу с подробной информацией карточки на основе её id. 
В компоненте CardLink реализована логика сохранения идентификатора карточки в localStorage с использованием JSON.parse и JSON.stringify при клике. 
Компонент History считывает историю просмотров из localStorage и извлекает параметр cardId из URL для последующей реализации динамического получения полной информации карточки через API.

suchachevskaya committed:
Рефакторинг: вынесение констант, сохранение истории и реализация перехода на страницу карточки

Константы навигационных элементов и маршрутов вынесены в отдельный модуль constants.ts, расположенный рядом с компонентом Header.
В компоненте CardLink реализована логика сохранения идентификатора карточки в localStorage с использованием JSON.parse и JSON.stringify при клике.
Создана логика перехода внутри карточки так, чтобы при клике осуществляется навигация на страницу с подробной информацией карточки на основе её id.
Компонент History считывает историю просмотров из localStorage и извлекает параметр cardId из URL для последующей реализации динамического получения полной информации карточки через API.

ElfistIO commented on src/components/Card/CardLink.tsx:
```diff
@@ -0,0 +1,25 @@
+  const navigate = useNavigate();
+  const handleSaveInHistory = () => {
+    const historyStorage = JSON.parse(
+      localStorage.getItem("cardHistory") || "[]"
```
ключи для работы со стором вынесите в константу, повторяются

ElfistIO commented on src/components/Card/CardLink.tsx:
```diff
@@ -0,0 +1,25 @@
+  };
+  const handleCardClick = () => {
+    handleSaveInHistory();
+    navigate(`/cards/${cardId}`);
```
/cards/ в ROUTES

ElfistIO commented on src/pages/CardPage/CardPage.tsx:
```diff
@@ -0,0 +1,55 @@
+import styles from "@/components/FavIcon/FavIcon.module.scss";
+import notFound from "@/assets/notFound.png";
+export function CardPage() {
+  const hgroupItem = [
```
вынести за компонент

ElfistIO commented on src/pages/CardPage/CardPage.tsx:
```diff
@@ -0,0 +1,55 @@
+    <div className="card-page">
+      <figure className="card-page__figure">
+        <div className="card-page__wrapper">
+          <img src={notFound} alt="" />
```
почему alt пустой?

suchachevskaya committed:
Рефакторинг

-Вынесены повторяющиеся части кода в константы
-Оптимизирован и доработан роутинг в компоненте CardLink с использованием generatePath.
-Исправлены мелкие недочёты, включая приведение типов и корректное использование ключей компонентов
